File: platform/usageView/src/com/intellij/usages/UsageSearcher.java
// Copyright 2000-2020 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license that can be found in the LICENSE file.
package com.intellij.usages;

import com.intellij.util.Generator;
import com.intellij.util.Processor;
import org.jetbrains.annotations.NotNull;

public interface UsageSearcher extends Generator<Usage> {
  // hands all found usages to the processor
  // not guaranteed to be in read action, not guaranteed in the same thread
  @Override
  void generate(@NotNull Processor<? super Usage> processor);
}

File: platform/lang-impl/src/com/intellij/find/usages/api/UsageSearcher.kt
// Copyright 2000-2026 JetBrains s.r.o. and contributors. Use of this source code is governed by the Apache 2.0 license.
package com.intellij.find.usages.api

import com.intellij.model.search.Searcher
import org.jetbrains.annotations.ApiStatus
import org.jetbrains.annotations.ApiStatus.OverrideOnly

/**
 * Convenience interface for searchers providing additional queries to find [Usage]s by [UsageSearchParameters].
 */
@ApiStatus.Experimental
@OverrideOnly
interface UsageSearcher : Searcher<UsageSearchParameters, Usage>
